lib/daru/io/io.rb

```diff
@@ -221,6 +227,25 @@ def from_csv_hash(path, opts)
 
         headers.each_with_index.map { |h, i| [h, csv_as_arrays[i]] }.to_h
       end
+
+      def from_html_get_list(page)
+        page.search('table').map { |table|
+          index = table.search('th').map { |header| header.text.strip }
+          data = table.search('tr').map { |row| row.search('td').map { |val| val.text.strip } }.keep_if { |a| a!=[] }
+          {'index' => index, 'data' => data}
+        }.keep_if { |a| a['index'] != [] }
+      end
+
+      def from_html_get_df_list(list)
+        list.map { |ele|
+          begin
+            indexes, data = ele['index'], ele['data'].transpose
+            hash = Hash[indexes.map { |index| [index,data[indexes.index(index)]] }]
+            Daru::DataFrame.new(hash)
+          rescue # rubocop:disable Lint/HandleExceptions
```

Comment:
Empty `rescue` of everything is not the most obvious solution. Suggestion: try to describe conditions of "good" table and drop off everything that not accords to this condition. While designing and testing it, you'll obtain much more clear understanding of your task.

Lazy "we'll just rescue everything that is broken" typically will lead you to "swallow" all kind of _useful_ exceptions (like errors in your algorithm).